### PR TITLE
fix #67 check if php path is executable

### LINKED ISSUE #67: Check if PHP Binary exist before launch
Step to reproduce:

1. Install PHP FPM and cli
2. Install Rymfony
3. perform `rymfony php:list --refresh`
4. uninstall PHP-FPM
5. perform `» RUST_BACKTRACE=1 RYMFONY_LOG=trace rymfony serve`

Result:

```
» RUST_BACKTRACE=1 RYMFONY_LOG=trace rymfony serve --passthru app_dev.php                                                                                                                                    
 2021-01-15T07:58:38.266Z  INFO > Starting PHP...
 2021-01-15T07:58:38.266Z TRACE > File /home/me/.rymfony/php-versions.json found
 2021-01-15T07:58:38.269Z DEBUG > PHP version set to 7.3 from ".php-version" file.
 2021-01-15T07:58:38.269Z TRACE > User selected version 7.3.26
 2021-01-15T07:58:38.269Z  INFO > Using php-fpm
 2021-01-15T07:58:38.269Z DEBUG > Running getuid
 2021-01-15T07:58:38.269Z DEBUG > Running getgid
thread 'main' panicked at 'Could not start php-fpm.', src/php/server_fpm.rs:136:5
stack backtrace:
   0: std::panicking::begin_panic
   1: rymfony::php::php_server::start
   2: rymfony::commands::serve::serve_foreground
   3: rymfony::commands::serve::serve
   4: rymfony::main
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

Suggested resolution:

Check if the PHP binary exist before launch. Exit with information to run the refresh PHP list command.

### PR DESCRIPTION
**Pull request by macintoshplus**
Fix #67 

Result:

image

### CODE REVIEW

File: src/php/php_server.rs [hunk 1]
```diff
@@ -32,6 +34,14 @@ impl PhpServer {
 pub(crate) fn start() -> PhpServer {
     let php_bin = binaries::get_project_version();
 
+    let phpbin_path = PathBuf::from(php_bin.as_str());
+
+    if !phpbin_path.is_executable() {
+        error!("PHP Binary not found or not executable: {}", php_bin.as_str());
+        info!("Execute rymfony php:list --refresh to solve this issue");
```
Comment (Pierstoval) [inline src/php/php_server.rs:41]:
```suggestion
        info!("You can execute \"rymfony php:list --refresh\" to update binaries paths cache");
```

File: src/php/php_server.rs [hunk 2]
```diff
@@ -32,6 +34,14 @@ impl PhpServer {
...
+        return PhpServer::new(0, PhpServerSapi::Unknown);
```
Comment (Pierstoval) [inline src/php/php_server.rs:42]: Shouldn't you just `panic!()` instead?

**Review by Pierstoval — CHANGES_REQUESTED** [review]

File: src/php/php_server.rs (hunk 2 above)
Comment (macintoshplus) [reply src/php/php_server.rs]: I have changed in `panic!`.

The previous behavior with the PR #69 permit Rymfony to start to serve only static file.

File: src/php/php_server.rs [hunk 3]
```diff
@@ -32,6 +34,14 @@ impl PhpServer {
 pub(crate) fn start() -> PhpServer {
     let php_bin = binaries::get_project_version();
 
+    let phpbin_path = PathBuf::from(php_bin.as_str());
+
+    if !phpbin_path.is_executable() {
+        error!("PHP Binary not found or not executable: {}", php_bin.as_str());
+        info!("You can execute \"rymfony php:list --refresh\" to update binaries paths cache");
```
Comment (Pierstoval) [inline src/php/php_server.rs:41]:
```suggestion
        error!("You can execute \"rymfony php:list --refresh\" to update binaries paths cache.");
```

For now, let's keep error messages together. An informational message after an error one doesn't really make sense.

In the future, we'll have to find a proper way to throw errors and have a look similar to Symfony CLI or Symfony Console.

**Review by Pierstoval — CHANGES_REQUESTED** [review]

**Pull request merged**